### DISCUSSION TITLE
HttpOnly Set-Cookie not working on Safari. Works with Chrome and Firefox.

### DISCUSSION THREAD

**Discussion by KentonParton** [fastapi-users/fastapi-users#959] [Q&A]
Tested an app in Safari for the first time today and discovered that the HTTP Only cookie isn't being set by the /callback route. The "Set-Cookie" property is present in the /callback response but there are no cookies present after the request. 

Subsequent requests, such as /user/me, do not include the cookie in the Request Header as is done for Chrome and Firefox.

I have tried the following configurations in the CookieTransport but without any success.

```python
cookie_transport = CookieTransport(
    cookie_name="swipleuserauth",
    cookie_max_age=settings.AUTH_LIFETIME_IN_SECONDS,
    cookie_domain=settings.BASE_URL
)
```

```python
cookie_transport = CookieTransport(
    cookie_name="swipleuserauth",
    cookie_max_age=settings.AUTH_LIFETIME_IN_SECONDS,
    cookie_samesite="None",
    cookie_domain=settings.BASE_URL
)
```

```python
cookie_transport = CookieTransport(
    cookie_name="swipleuserauth",
    cookie_domain=settings.BASE_URL
)
```

```python
cookie_transport = CookieTransport(
    cookie_name="swipleuserauth",
    cookie_max_age=settings.AUTH_LIFETIME_IN_SECONDS,
)
```

Here's a comparison between Firefox which is working and Safari which is not.

Firefox - WORKS

<img width="557" alt="Screenshot 2022-04-13 at 10 40 28" src="https://user-images.githubusercontent.com/20202312/163148966-aba53a16-0688-49b1-a5e9-3e0da56ce8bf.png">
<img width="556" alt="Screenshot 2022-04-13 at 10 40 36" src="https://user-images.githubusercontent.com/20202312/163149103-b2556dc3-ebb2-492f-8787-ef63b6177054.png">

Safari - DOES NOT WORK
<img width="495" alt="Screenshot 2022-04-13 at 10 42 56" src="https://user-images.githubusercontent.com/20202312/163149165-431e923c-cd37-4518-b7ee-a459a72bdc7a.png">
<img width="497" alt="Screenshot 2022-04-13 at 10 43 09" src="https://user-images.githubusercontent.com/20202312/163149216-80f7f2e0-d627-4ab7-9bec-75bf0754db7a.png">

I have disabled "Block all cookies" and "Prevent cross-site tracking" in Safari.

Has anyone run into this and know how to solve it? Appreciate the help!

**Comment by frankie567** [comment]
Hmm, I'm guessing it's because of the cookie secure flag. Have you tried this:

```py
cookie_transport = CookieTransport(
    cookie_name="swipleuserauth",
    cookie_secure=False,
)
```

I think that Firefox allows a `Secure` cookie to be set on localhost, even without SSL, for the convenience of developers. This is something Chrome was not doing before, but I noticed recently that it has now the same behavior as Firefox. Probably Safari is still conservative about this.
